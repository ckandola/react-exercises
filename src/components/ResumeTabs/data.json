[
    {
        "company": "Faithlife", 
        "dates": "October 2019 - January 2020",
        "duties": [
            "Using GIT.",
            "Learning ReactJS.", 
            "Learning to improve user interactions with React hooks", 
            "Worked on media editor features, including a QR Code menu, adding shortcuts, and modifying a color picker."
        ],
        "title": "Software Developer Intern"
    },
    {
        "company": "Cornerstone Systems NW", 
        "dates": "April 2019 - July 2019",
        "duties": [
            "Learning Javascript.", 
            "Learning DevTools.", 
            "Researching encryption algorithms included in Node.js Crypto module. Chose an algorithm based on security and speed requirements, created module for both encryption and decryption."
        ],
        "title": "Software Developer Intern"
    },
    {
        "company": "Sehome Diner", 
        "dates": "August 2012 - February 2020",
        "duties": [
            "Cashiering",
            "Greeting customers and taking orders",
            "Making espresso drinks and delivering food",
            "Cleaning tables, sweeping, washing dishes" 
        ],
        "title": "Waitress/Cashier"
    },
    {
        "company": "Lowe's", 
        "dates": "September 2019 - October 2019, January 2020 - present",
        "duties": [
            "Cashiering, although I also witnessed the transition to a new POS system, which had a LOT of issues." 
        ],
        "title": "Garden Cashier (present)"
    }
]
