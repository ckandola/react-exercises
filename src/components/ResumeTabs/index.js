// ** comment and uncomment for tutorial's data file vs my json file

import React, {useState/* , useEffect */} from 'react';
import { FaAngleDoubleRight} from 'react-icons/fa';
import './ResumeTabs.css';
import jobs from './data.json';

// const url = 'https://course-api.com/react-tabs-project'; **

const ResumeTabs = () => {
    // const [loading, setLoading] = useState(true);        **
    // const [jobs, setJobs] = useState([]);                **
    const [value, setValue] = useState(0);

    // const fetchJobs = async () => {                      ** --
    //     const response = await fetch(url);
    //     const newJobs = await response.json();
    //     setJobs(newJobs);
    //     setLoading(false);
    // };
    
    // useEffect(() => {
    //     fetchJobs();
    // }, []);                                              
    
    // if (loading) {
    //     return <section className="tab-section tab-loading">
    //         <h1>loading...</h1>
    //     </section>
    // }                                                    -- **

    const {company, dates, duties, title} = jobs[value];

    return (
        <section className="tab-section">
            <div className="tab-title">
                <h2>Professional Experience</h2>
                <div className="tab-underline"></div>
                <div className="tab-job-center">
                    <div className="tab-btn-container">
                        {jobs.map((job, index) => {
                            return (
                                <button 
                                    key={index} 
                                    className={`tab-job-btn ${index === value && 'tab-active-btn'}`}
                                    onClick={() => setValue(index)}>
                                        {job.title}
                                </button>
                            );
                        })}
                    </div>
                    <article className="tab-job-info">
                        <h3>{title}</h3>
                        <h4>{company}</h4>
                        <p className="tab-job-date">{dates}</p>
                        {duties.map((task, index) => {
                            return (
                                <div key={index} className="tab-job-desc">
                                    <FaAngleDoubleRight className="tab-job-icon"/>
                                    <p className="tab-job-duty">{task}</p>
                                </div>
                            );
                        })}
                    </article>
                </div>
            </div>
        </section >
    );
};

export default ResumeTabs;
