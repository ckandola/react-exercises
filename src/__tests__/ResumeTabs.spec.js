import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ResumeTabs from '../components/ResumeTabs';

const wrapper = shallow(<ResumeTabs />);

describe('ResumeTabs tests', () => {
    it('matches the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('displays a different job description when a different title is clicked', () => {
        const jobTitle = wrapper.find('.tab-active-btn').getElement().props.children;
        wrapper.find('.tab-job-btn').at(3).simulate('click');
        const newJob = wrapper.find('.tab-active-btn').getElement().props.children;
        expect(jobTitle).not.toBe(newJob);
    });
    
});
