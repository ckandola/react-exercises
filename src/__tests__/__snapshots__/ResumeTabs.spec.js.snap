// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResumeTabs tests matches the snapshot 1`] = `
<section
  className="tab-section"
>
  <div
    className="tab-title"
  >
    <h2>
      Professional Experience
    </h2>
    <div
      className="tab-underline"
    />
    <div
      className="tab-job-center"
    >
      <div
        className="tab-btn-container"
      >
        <button
          className="tab-job-btn tab-active-btn"
          key="0"
          onClick={[Function]}
        >
          Software Developer Intern
        </button>
        <button
          className="tab-job-btn "
          key="1"
          onClick={[Function]}
        >
          Software Developer Intern
        </button>
        <button
          className="tab-job-btn "
          key="2"
          onClick={[Function]}
        >
          Waitress/Cashier
        </button>
        <button
          className="tab-job-btn "
          key="3"
          onClick={[Function]}
        >
          Garden Cashier (present)
        </button>
      </div>
      <article
        className="tab-job-info"
      >
        <h3>
          Software Developer Intern
        </h3>
        <h4>
          Faithlife
        </h4>
        <p
          className="tab-job-date"
        >
          October 2019 - January 2020
        </p>
        <div
          className="tab-job-desc"
          key="0"
        >
          <FaAngleDoubleRight
            className="tab-job-icon"
          />
          <p
            className="tab-job-duty"
          >
            Using GIT.
          </p>
        </div>
        <div
          className="tab-job-desc"
          key="1"
        >
          <FaAngleDoubleRight
            className="tab-job-icon"
          />
          <p
            className="tab-job-duty"
          >
            Learning ReactJS.
          </p>
        </div>
        <div
          className="tab-job-desc"
          key="2"
        >
          <FaAngleDoubleRight
            className="tab-job-icon"
          />
          <p
            className="tab-job-duty"
          >
            Learning to improve user interactions with React hooks
          </p>
        </div>
        <div
          className="tab-job-desc"
          key="3"
        >
          <FaAngleDoubleRight
            className="tab-job-icon"
          />
          <p
            className="tab-job-duty"
          >
            Worked on media editor features, including a QR Code menu, adding shortcuts, and modifying a color picker.
          </p>
        </div>
      </article>
    </div>
  </div>
</section>
`;
